### PR TITLE
Improve package structure

### PR DESCRIPTION
**Pull request by manumerous**
This PR aims to address the following points to make it more accessible for other users and to eventually add it to PyPI:

- Introduce proper python package structure in the project 
- Add metadata and replace requirements.txt through pyproject.toml
- Give the project a shorter name
- Include type specifiers in functions declarations

Any feedback regarding those changes is very welcome. @Doeme @Yatekii @Jaeyoung-Lim 

However it still needs to be clarified how this new repo structure can be used as a submodule in another python project, without having to change all the imports to be relative. Any inputs,whether that makes sense or how that should be done are very welcome! As an example of this problem you can look at this project: 
https://github.com/ethz-asl/data-driven-dynamics/tree/master/Tools/parametric_model

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+name = "vpselector"
+version = "0.1.0"
+description = "Visualize and interactively select time-series data from a pandas DataFrame."
+readme = "README.md"
+requires-python = ">=3.8"
+license = {file = "LICENSE.txt"}
+keywords = ["python", "dataframe", "pandas", "visualization", "data selection", "time-series data", "data tools", "data science", "data-driven engineering", "machine learning", "system identification"]
+authors = [
+  {email = "manuel.galliker@gmx.ch"},
+  {name = "Manuel Yves Galliker"}
```
Comment (Yatekii) [inline pyproject.toml]:
```suggestion
"Manuel Yves Galliker <manuel.galliker@gmx.ch>"
```

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+]
+maintainers = [
+  {name = "Manuel Yves Galliker", email = "manuel.galliker@gmx.ch"}
```
Comment (Yatekii) [inline pyproject.toml]:
```suggestion
"Manuel Yves Galliker <manuel.galliker@gmx.ch>"
```

**Review by Yatekii — COMMENTED** [review]
Looking okay :)

**Comment by manumerous** [comment]
Thank you Noah! @Yatekii!

**Comment by manumerous** [comment]
Do you think it would be beneficial to add an app entrypoint? There would be an option to add that to the pyproject.toml file

**Comment by Yatekii** [comment]
I dont think its required for library usage :)

**Comment by manumerous** [comment]
Not even for the example?

**Comment by Yatekii** [comment]
Hmm I am not sure about that one ^^

**Comment by manumerous** [comment]
i am also not sure where that one best belongs. Maybe @Doeme has some insights there :)

**Comment by Doeme** [comment]
Depends on whether the example is a useful standalone program, e.g. just visualising a given pandas dataframe. If yes, you can add it as an entry point, if no, then don't. Also consider that entry points will be inside a PATH variable, i.e. they can be called via the command line as a global command.

**Comment by Yatekii** [comment]
Ok, then def. don't use entrypoints for examples =)

**Comment by manumerous** [comment]
Ok thanks for the feedback there. It is a standalone program that is hard coded to always use the test csv. So i think it should not be able to be accessed globally via command line. maybe having that functionality with a file selection dialog is something i could add in the future.

**Pull request merged**